src/components/AuthenticationProvider.tsx

```diff
@@ -33,20 +35,25 @@ export const AuthenticationProvider: ChildrenFC = ({ children }) => {
     }
   }, []);
 
-  const login = useCallback((newToken: string, newUserName: string) => {
-    localStorage.setItem(TOKEN_KEY, newToken);
-    setUserName(newUserName);
-  }, []);
+  const login = useCallback(
+    (newToken: string, newUserName: string, newUserEmail: string) => {
+      localStorage.setItem(TOKEN_KEY, newToken);
+      setUserName(newUserName);
+      setUserEmail(newUserEmail);
```

Comment:
Just set the whole user (without password, that must NOT be sent to the client) into one state.